lib/kasket/read_mixin.rb

```diff
@@ -49,6 +49,14 @@ def find_by_sql_with_kasket_on_id_array(keys)
 
     protected
 
+    def filter_pending_records(records)
+      if pending_records = Kasket.pending_records
+        records.map { |record| pending_records.fetch(record, record) }.compact
```

Comment:
Yeah, that's how I'm supporting destroyed records ... the key is the record for matching object identity, and the value is nil because it's been destroyed